### PR TITLE
[NEW] Livechat: avoid loopback in webhook events

### PR DESCRIPTION
**Pull request by hmagarotto**
Don't create webhook requests when the event was generated by an API call.

### CODE REVIEW

File: packages/rocketchat-livechat/imports/server/rest/messages.js [hunk 1]
```diff
@@ -46,7 +46,8 @@ RocketChat.API.v1.addRoute('livechat/messages', { authRequired: true }, {
 					_id: Random.id(),
 					rid,
 					token: visitorToken,
-					msg: message.msg
+					msg: message.msg,
+					receivedByAPI: true
```
Comment (renatobecker) [inline packages/rocketchat-livechat/imports/server/rest/messages.js]: What do you think on replacing `receivedByAPI: true` to `sender: 'API'`?

File: packages/rocketchat-livechat/server/hooks/sendToCRM.js [hunk 2]
```diff
@@ -77,6 +77,11 @@ RocketChat.callbacks.add('afterSaveMessage', function(message, room) {
 		return message;
 	}
 
+	// only call webhook if this message dont come from API
+	if (message.receivedByAPI) {
```
Comment (renatobecker) [inline packages/rocketchat-livechat/server/hooks/sendToCRM.js]: If we agree about `sender: 'API'`, you just need to replace here as well.

File: packages/rocketchat-livechat/imports/server/rest/messages.js (hunk 1 above)
Comment (hmagarotto) [reply packages/rocketchat-livechat/imports/server/rest/messages.js]: @renatobecker , that's better.
But, `sender` can cause confusion with the user that sent the message.
What if we use `channel: 'API'`?

Comment (renatobecker) [reply packages/rocketchat-livechat/imports/server/rest/messages.js]: @hmagarotto, Yes, I understand your point.. The problem using `channel` is that every `room` is a Rocket.Chat channel .. =/
Maybe `provider: 'API'`, what do you think?

Comment (hmagarotto) [reply packages/rocketchat-livechat/imports/server/rest/messages.js]: Good! I'll make the changes.

**Pull request closed without merging**